### ISSUE TITLE
Basic image trim with background

### ISSUE THREAD

**Issue by AlecRust** [libvips/ruby-vips#327]
I'm switching to VIPS from ImageMagick since Rails has made it the default in version 7, via [image_processing](https://github.com/janko/image_processing) gem.

I'm struggling with a basic "trim" to remove excess pixels (white in this case) around a product image:

![image](https://user-images.githubusercontent.com/247634/149532237-224940d2-6722-427f-af30-927f0285ff9e.png)

As mentioned in [this blog post](https://libvips.blogspot.com/2014/02/trim-auto-crop-with-ruby-vips.html) with ImageMagick these are the options to trim:

```ruby
{
  fuzz: '1%',
  trim: true
}
```

I'm not sure what the equivalent is with VIPS. From my reading I expected [`find_trim`](https://www.libvips.org/API/current/libvips-arithmetic.html#vips-find-trim) to work i.e.

```ruby
{
  find_trim: true
}

# OR

{
  find_trim: {
    threshold: 10
  }
}
```

But this doesn't work. Any ideas what I'm doing wrong?

I'd also like to remove the white background entirely in the case of logos i.e. replace white with transparent. With ImageMagick it's:

```ruby
{
  transparent: 'white'
}
```

But again I can find no VIPS equivalent for this. These seem like quite common transforms that should work with similar options?

**Comment by jcupitt** [comment]
Hi Alec, you'd need to ask image_processing (it's not my project), but at the CLI it'd be:

```
$ vips find_trim bottle.png 
95
20
61
233
```

Or in Ruby:

```
$ irb
irb(main):001:0> require "vips"
=> true
irb(main):002:0> x = Vips::Image.new_from_file "bottle.png"
=> #<Image 221x291 uchar, 4 bands, srgb>
irb(main):003:0> x.find_trim
=> [95, 20, 61, 233]
```

You can draw a box to visualize the rect with:

```
irb(main):004:0> x = x.draw_rect 0, *x.find_trim
=> #<Image 221x291 uchar, 4 bands, srgb>
irb(main):005:0> x.write_to_file "x.png"
=> nil
```

Making:

![x](https://user-images.githubusercontent.com/580843/149542931-e701e1de-00cc-4d04-b6c7-49c0affcb9c9.png)

**Comment by jcupitt** [comment]
Oh heh the github downsizer as made the 1px box invisible. Click on the image to see the rect.

**Comment by jcupitt** [comment]
.... sorry, last post, as I recall image_processing lets you drop down to ruby-vips as part of the chain of operations, so you'd need to add a line like:

```
x = x.crop *x.find_trim
```

somewhere. Have a look at the image_processing docs.

**Comment by AlecRust** [comment]
> Hi Alec, you'd need to ask image_processing

Yeah, I wondered if that was the right place but figured all that gem does is pass options through to VIPS.

Sounds like perhaps `image_processing` would need to add `find_trim` method to make it as easy as `{ find_trim: true }`, as it's [not mentioned](https://github.com/janko/image_processing/blob/master/doc/vips.md) in the docs.

Will open an issue there about this. Thanks for your useful reply anyway!

**Comment by kleisauke** [comment]
Closing in favor of https://github.com/janko/image_processing/issues/96.
